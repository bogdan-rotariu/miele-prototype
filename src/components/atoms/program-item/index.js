export { ProgramItem } from './program-item'
