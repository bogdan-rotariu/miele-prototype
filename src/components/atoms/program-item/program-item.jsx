import classNames from 'classnames'
import { string, func, bool, node } from 'prop-types'
import styles from './program-item.module.scss'

export const ProgramItem = ({ label, onClick, icon, subLabel, isActive }) => {
    if (!label || !icon) {
        return null
    }

    return (
        <button
            className={classNames(styles.container, {
                [styles.active]: isActive,
            })}
            onClick={onClick}
        >
            <span className={styles.icon}>{icon}</span>
            <span className={styles.label}>{label}</span>
            {subLabel && <span className={styles.subLabel}>{subLabel}</span>}
        </button>
    )
}

ProgramItem.propTypes = {
    label: string.isRequired,
    onClick: func.isRequired,
    icon: node.isRequired,
    subLabel: string,
    isActive: bool,
}
