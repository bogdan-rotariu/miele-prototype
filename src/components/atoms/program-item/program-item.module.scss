.container {
    align-items: center;
    background: none;
    border: 0;
    cursor: pointer;
    color: var(--color-white);
    display: flex;
    font-size: 1.25rem;
    font-family: var(--typography-roboto-font-family);
    flex-direction: column;
    flex: 1 0 auto;
    padding: 0;
    outline: 0;
    padding: 1rem 0;
}

.active {
    color: var(--color-primary);
}

.icon {
    font-size: 3.5rem;
    margin: 1rem 0;
}
