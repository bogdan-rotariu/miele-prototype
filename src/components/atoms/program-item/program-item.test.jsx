import { render } from '@testing-library/react'
import { ProgramItem } from './program-item'

describe('ProgramItem', () => {
    it('should match snapshot if render correctly', () => {
        const { container } = render(
            <ProgramItem
                label='Label'
                icon={() => ''}
                onClick={jest.fn()}
                subLabel='sub label'
            />,
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render nothing if missing label', () => {
        const { container } = render(
            <ProgramItem
                icon={() => ''}
                onClick={jest.fn()}
                subLabel='sub label'
            />,
        )
        expect(container.firstChild).toBeNull()
    })

    it('should render nothing if missing icon', () => {
        const { container } = render(
            <ProgramItem
                label='label'
                onClick={jest.fn()}
                subLabel='sub label'
            />,
        )
        expect(container.firstChild).toBeNull()
    })
})
