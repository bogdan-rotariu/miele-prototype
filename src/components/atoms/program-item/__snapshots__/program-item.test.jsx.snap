// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramItem should match snapshot if render correctly 1`] = `
<button
  class="container"
>
  <span
    class="icon"
  />
  <span
    class="label"
  >
    Label
  </span>
  <span
    class="subLabel"
  >
    sub label
  </span>
</button>
`;
